greedy_methods/fractional_knapsack.py

```diff
@@ -1,19 +1,38 @@
 from bisect import bisect
 from itertools import accumulate
 
+# Reference Link: https://en.wikipedia.org/wiki/Continuous_knapsack_problem
 
-def frac_knapsack(vl, wt, w, n):
+
+def frac_knapsack(
+    values: list[int], weights: list[int], capacity: int, max_items: int
+) -> float:
     """
+    This function implements fractional knapsack problem.
+
+    Args:
+        values: List of values of items.
+        weights: List of weights of items.
+        capacity: Capacity of the knapsack.
+        max_items: Number of items.
+
+    Returns:
+        Maximum value of items that can be put into the knapsack.
+
     >>> frac_knapsack([60, 100, 120], [10, 20, 30], 50, 3)
     240.0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 10, 4)
     105.0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, 4)
     95.0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6], 8, 4)
-    60.0
+    Traceback (most recent call last):
+        ...
+    ValueError: zip() argument 2 is shorter than argument 1
     >>> frac_knapsack([10, 40, 30], [5, 4, 6, 3], 8, 4)
-    60.0
+    Traceback (most recent call last):
+        ...
+    ValueError: zip() argument 2 is longer than argument 1
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 0, 4)
     0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, 0)
```

Comment:
Should we raise a `ValueError` if `capacity` is <= 0?  That would seem to me to be garbage data.

Our code does not consider `max_items` at all!